cassandra/metrics.go

```diff
@@ -5,45 +5,73 @@ import (
 	"strings"
 
 	"github.com/gocql/gocql"
-	"github.com/raintank/met"
+	"github.com/raintank/metrictank/stats"
 )
 
-type Metrics struct {
-	cassErrTimeout                  met.Count
-	cassErrTooManyTimeouts          met.Count
-	cassErrConnClosed               met.Count
-	cassErrNoConns                  met.Count
-	cassErrUnavailable              met.Count
-	cassErrCannotAchieveConsistency met.Count
-	cassErrOther                    met.Count
+type ErrMetrics struct {
+	cassErrTimeout                  *stats.Counter32
+	cassErrTooManyTimeouts          *stats.Counter32
+	cassErrConnClosed               *stats.Counter32
+	cassErrNoConns                  *stats.Counter32
+	cassErrUnavailable              *stats.Counter32
+	cassErrCannotAchieveConsistency *stats.Counter32
+	cassErrOther                    *stats.Counter32
 }
 
-func NewMetrics(component string, stats met.Backend) Metrics {
-	return Metrics{
-		cassErrTimeout:                  stats.NewCount(fmt.Sprintf("%s.error.timeout", component)),
-		cassErrTooManyTimeouts:          stats.NewCount(fmt.Sprintf("%s.error.too-many-timeouts", component)),
-		cassErrConnClosed:               stats.NewCount(fmt.Sprintf("%s.error.conn-closed", component)),
-		cassErrNoConns:                  stats.NewCount(fmt.Sprintf("%s.error.no-connections", component)),
-		cassErrUnavailable:              stats.NewCount(fmt.Sprintf("%s.error.unavailable", component)),
-		cassErrCannotAchieveConsistency: stats.NewCount(fmt.Sprintf("%s.error.cannot-achieve-consistency", component)),
-		cassErrOther:                    stats.NewCount(fmt.Sprintf("%s.error.other", component)),
+func NewErrMetrics(component string) ErrMetrics {
+	return ErrMetrics{
+
+		// metric idx.cassandra.error.timeout is a counter of timeouts seen to the cassandra idx
```

Comment:
we cleared this up on slack. but basically https://github.com/Dieterbe/metrics2docs is a bit simplistic, it can't parse the code very well, and as there are 2 places where we dynamically construct these metrics:
```
mdata/store_cassandra.go
68:    errmetrics = cassandra.NewErrMetrics("store.cassandra")

idx/cassandra/cassandra.go
48:    errmetrics           = cassandra.NewErrMetrics("idx.cassandra")
```

we need to add two comments for each metric here, and there needs to be an empty line between them because otherwise metrics2docs fails